cmd/webhook-server/main.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"flag"
+	"fmt"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"k8s.io/api/admission/v1beta1"
+	extapi "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"log"
+	"net/http"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	cephclusterResource = metav1.GroupVersionResource{Group: "ceph.rook.io", Version: "v1", Resource: "cephclusters"}
+	kubeClient          client.Client
+	scheme              = runtime.NewScheme()
+)
+
+func validateResource(req *v1beta1.AdmissionRequest) ([]patchOperation, error) {
+
+	var patches []patchOperation
+
+	//Checks if CephCluster is being updated and rejects the request if the dataDirHostPath is changed from initial value
+	if isCephClusterUpdate(req) {
+
+		//Fetch the existing ceph cluster object
+		found, err := fetchExistingCephCluster(req)
+		if err != nil {
+			return nil, err
+		}
+
+		raw := req.Object.Raw
+		updatedCephCluster := cephv1.CephCluster{}
+		//Deserializing the incoming CephCluster object
+		if _, _, err := universalDeserializer.Decode(raw, nil, &updatedCephCluster); err != nil {
+			return nil, fmt.Errorf("could not deserialize pod object: %v", err)
+		}
+
+		err = validateUpdatedCephCluster(updatedCephCluster, found)
+		if err != nil {
+			return nil, err
+		}
+
+	}
+
+	return patches, nil
+}
+
+func validateUpdatedCephCluster(updatedCephCluster cephv1.CephCluster, found cephv1.CephCluster) error {
```

Comment:
What about having the validation logic into its own file like `validate.go`?

Reply:
added validate.go for the further addition of validation logic